frontend/src/pages/notebookController/screens/server/AcceleratorSelectField.tsx

```diff
@@ -13,42 +13,47 @@ import {
 } from '@patternfly/react-core';
 import { isHTMLInputElement } from '~/utilities/utils';
 import { AcceleratorKind } from '~/k8sTypes';
-import SimpleDropdownSelect from '~/components/SimpleDropdownSelect';
-import useAccelerators from './useAccelerators';
+import SimpleDropdownSelect, { SimpleDropdownOption } from '~/components/SimpleDropdownSelect';
+import { AcceleratorState } from '~/pages/projects/screens/detail/notebooks/useNotebookAccelerator';
+import { UpdateObjectAtPropAndValue } from '~/pages/projects/types';
 import useAcceleratorCounts from './useAcceleratorCounts';
 
 type AcceleratorSelectFieldProps = {
-  accelerator?: AcceleratorKind;
-  setAccelerator: (accelerator?: AcceleratorKind) => void;
-  acceleratorCount?: number;
-  setAcceleratorCount: (size: number) => void;
+  acceleratorState: AcceleratorState;
+  setAcceleratorState: UpdateObjectAtPropAndValue<AcceleratorState>;
   supportedAccelerators?: string[];
   supportedText?: string;
 };
 
 const AcceleratorSelectField: React.FC<AcceleratorSelectFieldProps> = ({
-  accelerator,
-  setAccelerator,
-  acceleratorCount = 0,
-  setAcceleratorCount,
+  acceleratorState,
+  setAcceleratorState,
   supportedAccelerators,
   supportedText,
 }) => {
-  const [accelerators, loaded, loadError] = useAccelerators();
   const [detectedAcceleratorInfo] = useAcceleratorCounts();
 
+  const {
+    accelerator,
+    count: acceleratorCount,
+    accelerators,
+    useExisting,
+    additionalOptions,
+  } = acceleratorState;
   const validateAcceleratorCount = React.useCallback(
```

Comment:
Interesting -- I would have expected this to be a stand alone hook or to be a function you just provide the accelerator counts. 

This feels like something we can write tests for to make sure we are handling things. If your identifier is undefined, I think your use-cases start failing.

Reply:
it doesnt fail. If they have in improperly setup profile resource (how they got there idk). Then we will just say: `No accelerator detected with the identifier unknown`

Which is true, but we can have another edge case to just say that the resource is invalid. But then we should just filter those out from the beginning right?

Reply:
I converted it into a function that will run every render instead of a use effect